### DISCUSSION TITLE
Breaking changes from V10 -> V16 ?

### DISCUSSION THREAD

**Discussion by DoReVo** [nodejs/help#4968] [Q&A] [answered]
I am gathering information for breaking changes from Node 10 to Node 16, and as i was going through the [changelog](https://github.com/nodejs/node/blob/master/CHANGELOG.md), I couldn't find anything really. Am i correct to assume that there is not really any breaking changes going from 10 to 16? Or am i reading it wrong?

Background: I need to update a [loopback 3](https://loopback.io/doc/en/lb3/index.html) project from using node 10 to node 16. And was just checking for any breaking changes in node.

**Comment by BethGriggs** [comment, marked as answer]
There are many breaking changes between Node.js 10 to 16. To find them in the changelog, you'd need to review the semver major sections of all of the initial releases (vN.0.0) between Node.js 10 and 16, such as https://nodejs.org/en/blog/release/v11.0.0/#semver-major-commits, https://nodejs.org/en/blog/release/v12.0.0/#semver-major-commits, etc.

It may be easier to try to upgrade in steps versus jumping straight from Node.js 10 to 16, as you'd have a smaller subset of the breaking changes to review at each step.

**Reply by DoReVo** [reply]
I see now. Thanks !

**Reply by premaluu** [reply]
@BethGriggs Thanks for your suggestions, can you suggest which upgrade path would be better? like I want to go from 10 to 16 not directly, but let's say 10->12->16 or any other way?

**Reply by Trott** [reply]
If you don't want to or can't go directly from 10 to 16, using all the even numbered ones would be the best incremental way to go in my opinion: 10 -> 12 -> 14 -> 16

You can also try variations on that like only 10->12->16 or 10->14->16.

**Comment by UsboKirishima** [comment]
Surely if you want to upgrade your project from the node version 10 to the v16 you need to change few things.
By the way also the version 16 of nodejs is old, I suggest you to use a new version of nodejs.
You can look at the changelogs to replace your old code with the new features.

**Reply by Trott** [reply]
> By the way also the version 16 of nodejs is old,

No, it's not old. 16.x is the current LTS line. The most recent version was released on July 7, a few weeks ago. 16.x is the recommended version for production applications.
